DFReader.py

```diff
@@ -555,6 +555,103 @@ def set_message_timestamp(self, m):
         m._timestamp = self.timebase + count/rate
 
 
+class DFMetaData(object):
+    '''handle dataflash messages metadata'''
+    def __init__(self,parent):
+        self.parent = parent
+        self.data = None
+        self.tried = False
+
+    def reset(self):
+        '''clear cached data'''
+        self.data = None
+        self.tried = False
+
+    # Copy of function in mp_util, to avoid dependency on MavProxy
+    def dot_mavproxy(self,name=None):
+        '''return a path to store mavproxy data'''
+        if 'HOME' not in os.environ:
+            dir = os.path.join(os.environ['LOCALAPPDATA'], '.mavproxy')
+        else:
+            dir = os.path.join(os.environ['HOME'], '.mavproxy')
+        if name is None:
+            return dir
+        return os.path.join(dir, name)
+
+    def metadata_tree(self,verbose=False):
+        '''return a map between a log message and its metadata. May return None if data is not available'''
+        # If we've already tried loading data, use it if we have it
+        # This avoid repeated attempts, when the file is not there
+        if self.tried:
+            return self.data
+        self.tried = True
+        # Get file name, based on vehicle type
+        mapping = { mavutil.mavlink.MAV_TYPE_GROUND_ROVER : "Rover",
+                    mavutil.mavlink.MAV_TYPE_FIXED_WING : "Plane",
+                    mavutil.mavlink.MAV_TYPE_QUADROTOR : "Copter",
+                    mavutil.mavlink.MAV_TYPE_HELICOPTER : "Copter",
+                    mavutil.mavlink.MAV_TYPE_ANTENNA_TRACKER : "Tracker",
+                    mavutil.mavlink.MAV_TYPE_SUBMARINE : "Sub",
+                    mavutil.mavlink.MAV_TYPE_AIRSHIP : "Blimp",
+                    }
+        if self.parent.mav_type in mapping:
+            path = self.dot_mavproxy(os.path.join("LogMessages","%s.xml" % mapping[self.parent.mav_type]))
```

Comment:
I'll bring this up to tridge - perhaps we want a separate directory for pymavlink to put its metadata in, rather than re-using MAVProxy's.  That would entail some calls onto pymavlink to tell it to go and freshen bits of metadata, but would also mean that we're not leaning on the MAVProxy install to get pymavlink functionality.

Reply:
ok! I await direction!

Reply:
I can certainly see some advantages of having the download code on the pymavlink side....
- any script wanting to use could do so without it being pre-downloaded (`mylog.metadata.download()`).
- with `mavlogdump --verbose` there could be a `--download` or similar option to make sure that any enums are sure to be nicely populated.